packages/merge-styles/src/IStyleSet.ts

```diff
@@ -6,7 +6,18 @@ export type Diff<T extends keyof any, U extends keyof any> = ({ [P in T]: P } &
 
 export type Omit<U, K extends keyof U> = Pick<U, Diff<keyof U, K>>;
 
-export type TrimToFunction<T> = T extends (...args: any[]) => any ? T : () => T;
+export type __ReduceToFunction<T> = T extends (...args: any[]) => any ? T : never;
+
+/**
+ * Helper function whose role is supposed to express that regardless if T is a style object or style function,
+ * it will always map to a style function.
+ *
+ * Note: the commmented segment is the right expression but it is buggy in 2.8.2. Once Fabric upgrades to Typescript 3,
+ *  we should uncomment the commented segmment below as well as uncomment the test in mergeStyleSets.test.ts.
```

Comment:
I wonder if we should flag these with something we can search for later, like `__TS_2_8_WORKAROUND__` (which is inconsistently named already.)

Reply:
There is issue #6147 which explicitly tracks this todo, assigned to me. I think that is enough since I scrub my assigned items regularly but let me know if you strongly feel I should add the "search term" in the comment.

Reply:
Oops, I didn't see that comment earlier.